collections/_posts/2022-10-03-Swin-Transformer.md

```diff
@@ -0,0 +1,129 @@
+---
+layout: review
+title: "Swin Transformer: Hierarchical Vision Transformer using Shifted Windows"
+tags: deep-learning CNN transformer classification
+author: "Pierre Rougé"
+cite:
+    authors: "Ze Liu, Yutong Lin, Yue Cao, Han Hu, Yixuan Wei, Zheng Zhan, Stephen Lin, Baining Guo"
+    title:   "Swin Transformer: Hierarchical Vision Transformer using Shifted Windows"
+    venue:   " IEEE/CVF International Conference on Computer Vision 2021	"
+pdf: "https://arxiv.org/abs/2103.14030"
+---
+
+# Highlights
+
+* ViT has a quadratic complexity with respect to the number of tokens ( and therefore image size). This makes ViT unsuitable for dense prediction task requiring a huge number of tokens such as segmentation.
+* In the proposed architecture the Multi-Head Self Attention block (MSA) is replaced by a module based on **shifted windows** (W-MSA) which allow to obtain a **linear complexity** with respect to the number of token. 
+* Also in the ViT architecture the dimension of feature maps is fixed and constant
+* The proposed architecture build **hierarchical feature maps** by merging tokens in deeper layers
+
+
+# Architecture
+
+![](/collections/images/Swin-Transformer/architecture.jpg)
+
+* First the image of size $$H*W$$ is split in $$N$$ patches of size 4 x 4 to create tokens of size $$4*4*3=48$$
+
+  > ViT have bigger patches of size 16 x 16 ( so less suitable for dense prediction task such as segmentation)
+
+* Then a linear embedding layer is applied to project it to an arbitrary dimensions $$C$$ 
+
+* Two Swin Transfomer blocks (with modified self-attention) are applied, resulting in a new matrix $$\mathbf{z} \in \mathbb{R}^{N \times C}$$
+
+* To produce a hierarchical representation, the number of tokens is reduced by patch merging layers as the network gets deeper
+
+* First, features of each group of 2 x 2 neighboring patches are concatenates giving a matrix $$\mathbf{z} \in \mathbb{R}^{\frac{N}{4} \times 4C}$$
+
+* A linear layer is applied to set the dimension of tokens to 2C and Swin Transformers blocks are applied
+
+* This first block of patch merging is referred as Stage 2
+
+* This process is repeated in Stage 3 ans Stage 4 to obtain a final matrix $$\mathbf{z} \in \mathbb{R}^{\frac{N}{64} \times 8C}$$
+
+  > These steps produce a hierarchical representation similar to a convolutional networks. Therefore Swin Transformers can serve as a general backbone for several vision tasks.
+
+
+# Shifted Window based Self-Attention
+
+* Swin Transformer block is built by replacing the standard multi-head self attention (MSA) module in a Transformer block by a module based on  **shifted windows**
+
+  ![](/collections/images/Swin-Transformer/transformer-block.jpg)
```

Comment:
An additional remark: just precise what LN means in the figure (is it layer normalization ?)

Reply:
Yes, it stands for layer normalization, I added the precision in the post ! :)